codalab/server/bundle_manager.py

```diff
@@ -412,7 +440,7 @@ def _schedule_run_bundles_on_workers(self, workers, staged_bundles_to_run, user_
         # To avoid the potential race condition between bundle manager's dispatch frequency and
         # worker's checkin frequency, update the column "exit_after_num_runs" in worker table
         # before bundle manager's next scheduling loop
-        for worker in workers_list:
+        for worker in workers.workers():
```

Comment:
@candicegjing this is an unrelated fix, but I noticed that `workers_list` is essentially just the workers corresponding to the _last_ bundle, not all workers (as I think you intended it). Maybe you can check and verify this makes sense?

Reply:
I think `workers_list` is what I want here because `workers_list` contains the minimum number of workers that would need to update their `exit_after_num_runs` value in the database (it's the list of eligible workers that could potentially run the current bundle).  Plus, `workers.workers()` doesn't reflect the changes of `exit_after_num_runs` for each worker, as we were calculating on a deep copy of the original object. But let me know if I misunderstood...

Reply:
> I think workers_list is what I want here because workers_list contains the minimum number of workers that would need to update their exit_after_num_runs value in the database (it's the list of eligible workers that could potentially run the current bundle)

oh, got it. I think you meant to indent this for loop by 4, then? Right now, it runs after all of the bundles are finished scheduling.